libs/ast_generic/AST_generic.ml

```diff
@@ -273,7 +273,7 @@ module IdInfoId = Gensym.MkId ()
  *)
 (* a single unique gensym'ed number. *)
 type sid = SId.t
-and resolved_name = resolved_name_kind * sid
+and resolved_name = resolved_name_kind * sid [@@deriving show, eq]
```

Comment:
you should not add @@deriving in the middle of a mutually recursive type.
There is already a deriving show and eq further below. There is no need for this one.